### PR TITLE
Persist position in helpers and code tabs

### LINKED ISSUE #1220: Persist cursor position when switching between tabs in editor

### PR DESCRIPTION
**Pull request by flaurida**
* resolves #1220

### CODE REVIEW

File: app/components/Editor/Sidebar/HelpersEditor.tsx [hunk 1]
```diff
@@ -45,13 +45,18 @@ export default function HelpersEditor({ onKeyDown }: Props): JSX.Element {
...
+        editor.setPosition({ column: 1, lineNumber: helpersLine });
+      }
     }
 
     // update current team helpers version so it works in callback
     if (team.helpers_version > -1) {
       helpersVersionRef.current = team.helpers_version;
     }
-  }, [editor, team]);
+  }, [editor, helpersLine, team]);
```
Comment (flaurida) [inline app/components/Editor/Sidebar/HelpersEditor.tsx:59]: I don't want to re-run the hook every time this changes, safe to ignore `helpersLine` as a dependency?

File: app/components/Editor/Sidebar/CodeEditor/index.tsx [hunk 2]
```diff
@@ -28,14 +34,24 @@ export default function CodeEditor({ onKeyDown }: Props): JSX.Element {
...
+    );
   };
 
   useEffect(() => {
     if (!controller || !editor) return;
 
     // set the initial editor code
     editor.setValue(controller.code);
+    // restore position
+    if (codeLine) {
```
Comment (flaurida) [inline app/components/Editor/Sidebar/CodeEditor/index.tsx:51]: likewise not sure `codeLine` should be a dependency of `useEffect`

Comment (flaurida) [reply app/components/Editor/Sidebar/CodeEditor/index.tsx:51]: (if it is it keeps setting the editor value)

**Review by jperl — CHANGES_REQUESTED** [review]
Going to submit a different PR with a simpler approach, just don't remove the elements from the dom when switching tabs

**Pull request closed without merging**